src/utils/stark.ts

```diff
@@ -40,26 +42,32 @@ export function makeAddress(input: string): string {
   return addHexPrefix(input).toLowerCase();
 }
 
-export function formatSignature(sig?: SignatureType): string[] {
-  if (!sig) return [];
+export function formatSignature(sig?: AnySignatureType): ArraySignatureType {
+  if (!sig) throw Error('formatSignature: provided signature is undefined');
+  if (typeof sig === 'string' || typeof sig === 'number') {
+    return [toHex(sig)];
+  }
+  if (Array.isArray(sig)) {
+    return sig.map((it) => toHex(it));
+  }
   try {
     const { r, s } = sig;
     return [toHex(r), toHex(s)];
   } catch (e) {
-    return [];
+    throw new Error('Signature need to be WeierstrassSignatureType or an array');
   }
 }
 
-export function signatureToDecimalArray(sig?: SignatureType): string[] {
+export function signatureToDecimalArray(sig?: AnySignatureType): ArraySignatureType {
   return bigNumberishArrayToDecimalStringArray(formatSignature(sig));
 }
 
-export function signatureToHexArray(sig?: SignatureType): string[] {
+export function signatureToHexArray(sig?: AnySignatureType): ArraySignatureType {
   return bigNumberishArrayToHexadecimalStringArray(formatSignature(sig));
 }
 
-export function parseSignature(sig?: string[]) {
-  if (!sig) return undefined;
+export function parseSignature(sig?: ArraySignatureType): WeierstrassSignatureType {
+  if (!sig) throw Error('parseSignature: provided signature is undefined');
 
   const [r, s] = sig;
   return new Signature(toBigInt(r), toBigInt(s));
```

Comment:
This would not work if `sig = [r1, s1, r2, s2 ]`. I would suggest in this case to return `WeierstrassSignatureType[]` or get rid of this function completely.

Reply:
Yea this is not flexible at all, this is used only in parseGetTransactionResponse, but I am not sure this response will return a signature anymore, maybe if you query some old transactions.
I am in favor of dropping it and just returning raw data as it is in that case.

Reply:
That sounds okay to me